cli/register/register.go

```diff
@@ -241,6 +242,11 @@ func DoIt(org, pattern, nodeIdTok, userPw, inputFile string, nodeOrgFromFlag str
 				if n.HAGroup != "" && nodeType == persistence.DEVICE_TYPE_CLUSTER {
 					cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("The node %v is in the HA group %v but the agent is 'cluster' type. HA group does not support 'cluster' type members. Please remove the node from the HA group or choose a different node id with -n flag.", nId, n.HAGroup))
 				}
+
+				// Check clusterNamesapce
+				if nodeType == persistence.DEVICE_TYPE_CLUSTER {
+					clusterNamesapce = SetClusterNamespaceInExchange(org, nodeId, cliutils.OrgAndCreds(userOrg, userAuth), n)
+				}
```

Comment:
Setting cluster namespace should be done inside the agent.

Reply:
Good place is here https://github.com/open-horizon/anax/blob/master/api/path_node.go#L245 when the agent is first associated with the exchange node.